### PR TITLE
PWN-8638 - Add striga KYC values repositories

### PR DESCRIPTION
**Pull request by gslevinkov**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-8638

## Description of Work

occupation, source of funds, countries with emodji

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/auth/repository/Country.kt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package org.p2p.wallet.auth.repository
+
+class Country(
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/auth/repository/Country.kt:3]: maybe WorldCountry? Or GlobalCountry? Country looks very simple..

File: app/src/main/java/org/p2p/wallet/striga/repository/StrigaKycDataLocalRepository.kt [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package org.p2p.wallet.striga.repository
+
+import org.p2p.wallet.striga.repository.model.StrigaOccupation
+import org.p2p.wallet.striga.repository.model.StrigaSourceOfFunds
+
+interface StrigaKycDataLocalRepository {
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/striga/repository/StrigaKycDataLocalRepository.kt:6]: how's kyc data sounds? it is too ambiguous?

Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/striga/repository/StrigaKycDataLocalRepository.kt:6]: maybe StrigaValuesLocalRepo?

File: app/src/main/java/org/p2p/wallet/auth/ui/phone/CountryCodeInteractor.kt
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/auth/ui/phone/CountryCodeInteractor.kt:1]: deleted due to just proxying methods

File: app/src/main/java/org/p2p/wallet/auth/repository/Country.kt (hunk 1 above)
Comment (eduardmaximovich) [reply app/src/main/java/org/p2p/wallet/auth/repository/Country.kt:3]: Country is ok

File: app/src/main/java/org/p2p/wallet/auth/repository/CountryInMemoryRepository.kt [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+class CountryInMemoryRepository(
+    private val countryCodeRepository: CountryCodeLocalRepository,
+) : CountryRepository {
+    override suspend fun getAllCountries(): List<Country> {
+        return countryCodeRepository.getCountryCodes()
+            .map { it.extractCountry() }
+    }
+
+    override suspend fun detectCountryOrDefault(): Country {
+        val countryCodeByDetekt = countryCodeRepository.detectCountryCodeBySimCard()
```
Comment (eduardmaximovich) [inline app/src/main/java/org/p2p/wallet/auth/repository/CountryInMemoryRepository.kt:14]: maybe `detectedCode`? and there's a typo: countryCodeByDete[c]t

File: app/src/main/java/org/p2p/wallet/striga/repository/StrigaKycDataLocalRepository.kt (hunk 2 above)
Comment (eduardmaximovich) [reply app/src/main/java/org/p2p/wallet/striga/repository/StrigaKycDataLocalRepository.kt:6]: StrigaValuesLocalRepo or even StrigaPresetDataLocalRepo sounds better, as this repo doesn't actually contains any KYC-specific data, it's just a dataset

File: app/src/main/java/org/p2p/wallet/auth/repository/CountryInMemoryRepository.kt (hunk 3 above)
Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/auth/repository/CountryInMemoryRepository.kt:14]: I am full in Detekt, thanks

**Pull request merged**